File: jambeth/jambeth-security/src/main/java/de/osthus/ambeth/security/ISecurityActivation.java
package de.osthus.ambeth.security;

import de.osthus.ambeth.threading.IResultingBackgroundWorkerDelegate;

public interface ISecurityActivation
{
	boolean isSecured();

	boolean isFilterActivated();

	<R> R executeWithoutSecurity(IResultingBackgroundWorkerDelegate<R> pausedSecurityRunnable) throws Throwable;

	<R> R executeWithoutFiltering(IResultingBackgroundWorkerDelegate<R> noFilterRunnable) throws Throwable;
}

File: ambeth/Ambeth.Merge/ambeth/security/ISecurityActivation.cs
using De.Osthus.Ambeth.Threading;

namespace De.Osthus.Ambeth.Security
{
    public interface ISecurityActivation
    {
	    bool Secured { get;  }

		bool ServiceSecurityEnabled { get; }

        bool FilterActivated { get; }

		void ExecuteWithSecurityDirective(SecurityDirective securityDirective, IBackgroundWorkerDelegate runnable);

		R ExecuteWithSecurityDirective<R>(SecurityDirective securityDirective, IResultingBackgroundWorkerDelegate<R> runnable);
		
	    R ExecuteWithoutSecurity<R>(IResultingBackgroundWorkerDelegate<R> pausedSecurityRunnable);

	    R ExecuteWithoutFiltering<R>(IResultingBackgroundWorkerDelegate<R> noFilterRunnable);

		R ExecuteWithFiltering<R>(IResultingBackgroundWorkerDelegate<R> filterRunnable);
    }
}
